File: AndroidParty/app/src/main/java/com/tesonet/justas/androidparty/ServerListAdapter.java
package com.tesonet.justas.androidparty;

import android.content.Context;
import android.util.Pair;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import java.util.List;

public class ServerListAdapter extends BaseAdapter {
    private final Context mContext;
    private List<Pair<String, String>> mServerList;
    public ServerListAdapter(Context context, List<Pair<String, String>> serverList) {
        mContext = context;
        mServerList = serverList;
    }

    static class ViewHolder {
        public TextView country;
        public TextView distance;
    }

    @Override
    public int getCount() {
        return mServerList.size();
    }

    @Override
    public Object getItem(int position) {
        return mServerList.get(position);
    }

    @Override
    public long getItemId(int position) {
        return mServerList.get(position).hashCode();
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        ViewHolder holder;
        if (convertView == null) {

            convertView = LayoutInflater.from(mContext).inflate(R.layout.list_item, parent, false);
            holder = new ViewHolder();
            holder.country = convertView.findViewById(R.id.list_item_country);
            holder.distance = convertView.findViewById(R.id.list_item_distance);

            convertView.setTag(holder);

        } else {
            holder = (ViewHolder) convertView.getTag();
        }

        Pair<String, String> item = mServerList.get(position);

        holder.country.setText(item.first);
        String distanceInKm = item.second + " km";
        holder.distance.setText(distanceInKm);

        return convertView;
    }
}

File: app/src/main/java/com/example/androidparty/ServerListAdapter.kt
package com.example.androidparty

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.example.androidparty.db.ServerEntity
import kotlinx.android.synthetic.main.server_item_layout.view.*

class ServerListAdapter(private val context: Context) :
    RecyclerView.Adapter<ServerListAdapter.ServerViewHolder>() {

    var serverList: List<ServerEntity> = emptyList()

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ServerViewHolder {
        val inflatedView =
            LayoutInflater.from(context).inflate(R.layout.server_item_layout, parent, false)
        return ServerViewHolder(inflatedView)
    }

    fun setList(serverList: List<ServerEntity>) {
        this.serverList = serverList
    }

    override fun getItemCount() = serverList.size

    override fun onBindViewHolder(holder: ServerViewHolder, position: Int) {
        holder.bind(serverList[position])
    }

    class ServerViewHolder(val view: View) : RecyclerView.ViewHolder(view) {

        fun bind(serverData: ServerEntity) {
            view.viewTextViewCountry.text = serverData.name
            view.viewTextViewDistance.text = String.format("%s km", serverData.distance)
        }
    }
}